tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,458 @@
+"""Model setup for the fracturized pressure problem using Sneddon's analytical solution.
+
+References:
+
+    - [1] Starfield, C.: Boundary Element Methods in Solid Mechanics (1983)
+    - [2] Sneddon, I.N.: Fourier transforms (1951)
+
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+import numpy as np
+
+import porepy as pp
+from porepy.applications.convergence_analysis import ConvergenceAnalysis
+from porepy.applications.md_grids.model_geometries import (
+    SquareDomainOrthogonalFractures,
+)
+from porepy.geometry.distances import point_pointset
+
+
+class SneddonExactSolution2d:
+    """Class representing the analytical solution for the pressurized fracture problem."""
+
+    def __init__(self, model: "ManuSneddonSetup2d"):
+        self.p0 = model.params["p0"]
+        self.theta_rad = model.params["theta_rad"]
+
+        self.a = model.params["a"]
+        self.shear_modulus = model.params["material_constants"]["solid"].shear_modulus
+        self.poi = model.params["poi"]
+        self.n = model.params["num_bem_segments"]
+        self.length = model.domain_size
+
+    def exact_sol_global(self, sd: pp.Grid) -> np.ndarray:
+        """Compute the analytical solution for the pressurized fracture problem in
+        question.
+
+        Parameters:
+            sd: Subdomain for which the analytical solution is to be computed.
+
+        """
+
+        h = 2 * self.a / self.n
+        box_faces = sd.get_boundary_faces()
+        u_bc = np.zeros((sd.dim, sd.num_faces))
+
+        center = np.array([self.length / 2, self.length / 2, 0])
+        bem_centers = self.get_bem_centers(h, center)
+        eta = point_pointset(bem_centers, center)
+
+        u_a = -self.analytical_displacements(eta)
+        u_bc = self.displacement_bc_values_from_bem(
+            sd, h / 2, box_faces, bem_centers, u_a
+        )
+        return u_bc
+
+    def exact_sol_fracture(self, mdg: pp.MixedDimensionalGrid) -> np.ndarray:
```

Comment:
Can't we remove argument, since the class has mdg as an attribute?

Reply:
The exact solution is actually an attribute of the model class, not a mixin.
We could make it save a reference to the mdg upon instantiation as an alternative, but so far it saves only numerical parameters there.

Reply:
Right. Keep as is, then!